src/Protocol/Protocol18x.cpp

```diff
@@ -1834,39 +1834,40 @@ void cProtocol180::AddReceivedData(const char * a_Data, size_t a_Size)
 		}
 		
 		// Check packet for compression:
-		UInt32 CompressedSize = 0;
+		UInt32 UncompressedSize = 0;
 		AString UncompressedData;
 		if (m_State == 3)
 		{
 			UInt32 NumBytesRead = static_cast<UInt32>(m_ReceivedData.GetReadableSpace());
-			m_ReceivedData.ReadVarInt(CompressedSize);
-			if (CompressedSize > PacketLen)
-			{
-				m_Client->Kick("Bad compression");
-				return;
-			}
-			if (CompressedSize > 0)
+			m_ReceivedData.ReadVarInt(UncompressedSize);
```

Comment:
I think we shouldn't assume, because malicious clients could send a malformed packet with `PacketLength` set to, say, 1, but `UncompressedSize` set to 800000; this code would then probably fail.